### PR TITLE
fix(testing): switch test_hydra_sweep_ddp_sim to integer batch limits to avoid DDP sharding to 0

### LINKED ISSUE #626: fix(testing): test_hydra_sweep_ddp_sim — limit_val_batches=0.1 * DDP processes yields 0 batches (same as #46)
## Problem

`tests/test_sweeps.py::test_hydra_sweep_ddp_sim` fails end-to-end under DDP sim on GPU CI because `+trainer.limit_val_batches=0.1` combined with the DDP sharding of a small validation set produces 0 batches on each rank. Lightning rejects this as a `MisconfigurationException` before `trainer.fit` can make progress.

## Root cause

Lightning shards the val loader across DDP ranks before applying the `limit_val_batches` fraction. With a small val loader (5 batches) and a fraction of `0.1`, the per-rank slice rounds to 0 batches:

```
lightning.fabric.utilities.exceptions.MisconfigurationException:
  You requested to check 0.1 of the `val_dataloader` but 0.1 * 5 < 1.
  Please increase the `limit_val_batches` argument. Try at least `limit_val_batches=0.2`.
```

The sweep test sets all three fraction-based limits at `tests/test_sweeps.py:70-72`:

```python
"+trainer.limit_train_batches=0.01",
"+trainer.limit_val_batches=0.1",
"+trainer.limit_test_batches=0.1",
```

## Sibling fix

This is the same class of bug as #46, which was fixed for `test_train_ddp_sim` in #515 by switching the fraction limits to integer limits (e.g. `limit_val_batches=1`). The fix template was never carried over to the sweep test in `tests/test_sweeps.py`.

Relevant snippet from PR #515 (applied to `test_train_ddp_sim`):

```python
# Integer limits avoid the `fraction * batches < 1` error when the
# fixture's fractions (0.01, 0.1) shrink too far under DDP sharding.
cfg_train.trainer.limit_train_batches = 1
cfg_train.trainer.limit_val_batches = 1
cfg_train.trainer.limit_test_batches = 1
```

## Evidence

Level 1 verification of #621 (which fixed the `trainer.max_epochs` struct-mode error) showed that the sweep now launches and DDP-sim initializes both processes, but training then fails on the `limit_val_batches` issue. See the Level 1 verification comment on PR #621: https://github.com/tinaudio/synth-setter/pull/621#issuecomment-4281310366

This issue is a sibling to #620 (also tracking a test_hydra_sweep_ddp_sim failure surfaced by #621's verification).

## Acceptance criteria

- [ ] `tests/test_sweeps.py::test_hydra_sweep_ddp_sim` passes end-to-end on GPU CI after switching the three fraction limits to integer limits.
- [ ] The non-DDP `test_hydra_sweep` is left untouched (its fractions work fine without DDP sharding).
- [ ] The skipped tests (`test_optuna_sweep`, `test_optuna_sweep_ddp_sim_wandb`) are left untouched — #514 owns those rewrites.

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

- Switch `tests/test_sweeps.py::test_hydra_sweep_ddp_sim` from fraction-based limits (`0.01` / `0.1` / `0.1`) to integer limits (`1` / `1` / `1`).
- Lightning shards val/test loaders across DDP ranks before applying `limit_*_batches`. With `limit_val_batches=0.1` on a 5-batch val loader under DDP sim, the per-rank slice rounds to 0 batches and Lightning raises `MisconfigurationException` before `trainer.fit` can progress.
- This is a sibling fix to #515, which applied the same integer-limits template to `test_train_ddp_sim` for the same root cause (#46). The template was never carried over to the sweep test.
- `test_hydra_sweep` (non-DDP) is intentionally left untouched — its fractions work fine without DDP sharding. The skipped `test_optuna_sweep*` tests are also left alone; #514 owns their rewrite.

Fixes #626

## Test plan

- [ ] GPU CI runs `pytest tests/test_sweeps.py::test_hydra_sweep_ddp_sim -v -m "slow and gpu"` and it passes end-to-end (cannot be verified locally without GPU + DDP sim).
- [x] `make format` passes cleanly on the changed file.

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

This is a test-only change that flips three Hydra sweep overrides from fraction (`0.01` / `0.1` / `0.1`) to integer (`1`). The only true end-to-end proof lives behind `@pytest.mark.gpu` + `@pytest.mark.slow`, so the Level 1 check is deferred to GPU CI and documented here with the exact command to reproduce.

- [ ] **`test_hydra_sweep_ddp_sim` passes end-to-end on GPU + DDP sim** *(Level 1 — SKIP: requires `@pytest.mark.gpu` hardware + 2-process DDP sim; the environment this verification ran in has no GPU. This is the verification that actually proves the fix.)*
  To verify on a machine with a CUDA GPU:
  ```bash
  git fetch origin fix/test-hydra-sweep-ddp-integer-limits
  git worktree add /tmp/verify-629 origin/fix/test-hydra-sweep-ddp-integer-limits
  cd /tmp/verify-629
  pytest tests/test_sweeps.py::test_hydra_sweep_ddp_sim -v -m "slow and gpu" --no-header
  ```
  Expected: exit 0, no `MisconfigurationException: You requested to check 0.1 of the val_dataloader but 0.1 * 5 < 1` from Lightning.

- [x] **Test file still collects cleanly (no syntax/import regression from the edit)** *(Level 2 — invoked pytest's collector directly)*
  ```bash
  $ python3 -m pytest tests/test_sweeps.py --collect-only -q 2>&1 | tail -15
  tests/test_sweeps.py::test_experiments
  tests/test_sweeps.py::test_hydra_sweep
  tests/test_sweeps.py::test_hydra_sweep_ddp_sim
  tests/test_sweeps.py::test_optuna_sweep
  tests/test_sweeps.py::test_optuna_sweep_ddp_sim_wandb

  5 tests collected in 0.39s
  ```

- [x] **Runtime-constructed Hydra overrides are integer-valued, not fractions** *(Level 2 — parsed via AST and asserted `val.isdigit()` on every `limit_*_batches` override, which is stronger than diff-grep: it proves the values the subprocess will actually receive)*
  ```bash
  $ python3 -c "
  import ast
  src = open('tests/test_sweeps.py').read()
  tree = ast.parse(src)
  target = next(n for n in ast.walk(tree) if isinstance(n, ast.FunctionDef) and n.name == 'test_hydra_sweep_ddp_sim')
  cmd_assign = next(s for s in target.body if isinstance(s, ast.Assign) and s.targets[0].id == 'command')
  list_node = cmd_assign.value.left
  limits = [ast.literal_eval(e) for e in list_node.elts if isinstance(e, ast.Constant) and 'limit_' in e.value]
  print('Limit overrides found:', limits)
  for L in limits:
      key, val = L.split('=')
      assert val.isdigit(), f'Expected integer limit, got {val!r} in {L!r}'
  print('All three limits are integer-valued.')
  "
  Limit overrides found: ['+trainer.limit_train_batches=1', '+trainer.limit_val_batches=1', '+trainer.limit_test_batches=1']
  All three limits are integer-valued.
  ```

- [x] **Ruff passes on the edited file** *(Level 2 — invoked the linter directly; stronger than trusting `make format` because it runs against the post-edit file in isolation)*
  ```bash
  $ /home/dev/.cache/pre-commit/repohofkwexw/py_env-python3/bin/ruff check tests/test_sweeps.py
  All checks passed!
  ```

- [x] **No stale fraction references to the targeted test** *(adversarial — checked that the edit did not leave an inconsistent state where another file references the old values for `test_hydra_sweep_ddp_sim`)*
  ```bash
  $ rg 'limit_val_batches=0\.1|limit_train_batches=0\.01|limit_test_batches=0\.1'
  tests/test_sweeps.py:120:        "+trainer.limit_train_batches=0.01",
  tests/test_sweeps.py:121:        "+trainer.limit_val_batches=0.1",
  tests/test_sweeps.py:122:        "+trainer.limit_test_batches=0.1",
  CHANGELOG.md:512:The fixture's limit_val_batches=0.1 yielded 0 val batches per rank under ddp_spawn with devices=2
  ```
  Remaining hits are intentional:
  - `tests/test_sweeps.py:120-122` is `test_optuna_sweep_ddp_sim_wandb`, which is `@pytest.mark.skip` and owned by #514 — the task explicitly scoped this PR to only `test_hydra_sweep_ddp_sim`.
  - `CHANGELOG.md:512` documents the historical #515 fix; not a live reference.

- [x] **Scope guard: sibling tests untouched** *(adversarial — `test_hydra_sweep` (non-DDP) must keep its `fast_dev_run=true` + no-limits shape, which the task required)*
  ```bash
  $ git diff origin/main -- tests/test_sweeps.py | grep -E '^[+-]' | grep -v '^[+-]\{3\}'
  -        "+trainer.limit_train_batches=0.01",
  -        "+trainer.limit_val_batches=0.1",
  -        "+trainer.limit_test_batches=0.1",
  +        "+trainer.limit_train_batches=1",
  +        "+trainer.limit_val_batches=1",
  +        "+trainer.limit_test_batches=1",
  ```
  Only the three targeted lines changed; `test_hydra_sweep`, `test_experiments`, `test_optuna_sweep`, and `test_optuna_sweep_ddp_sim_wandb` are byte-identical to `origin/main`.

**Comment by ktinubu** [comment]
**Stacking note for reviewers**: this PR is based on `main` but the test it fixes also depends on #621's `+trainer.max_epochs=3` force-add fix (still open). Three merge orderings work:

1. **Merge #621 first, then rebase this PR on main** — cleanest; the fraction-limit fix is an independent concern.
2. **Merge this PR first, then #621** — test still fails end-to-end on CI until #621 lands, but the diff is self-contained.
3. **Squash both into one PR** — if preferred, close this and move the integer-limits change into #621.

No preference from my side; flagging so whoever merges isn't surprised that `test_hydra_sweep_ddp_sim` only goes green once both land.
